ext/native/util/text/utf8.cpp

```diff
@@ -477,12 +477,69 @@ std::string ConvertUCS16ToUTF8(const std::u16string &wstr) {
 	return s;
 }
 
-static size_t ConvertUTF8ToUCS16Internal(char16_t *dest, size_t destSize, const std::string &source) {
+static size_t ConvertUTF8ToUCS2Internal(char16_t *dest, size_t destSize, const std::string &source) {
 	const char16_t *const orig = dest;
 	const char16_t *const destEnd = dest + destSize;
 
 	UTF8 utf(source.c_str());
 
+	char16_t *destw = (char16_t *)dest;
+	const char16_t *const destwEnd = destw + destSize;
+
+	// TODO: Instead of encoding UTF16 here we should encode UCS2.
```

Comment:
Note: they are the same as long as it's within BMP.  You can encode UCS-2 by encoding to UTF-16 and failing/ignoring any 16-bit unit produced by UTF-16 in the range U+D800 — U+DFFF.

-[Unknown]

Reply:
Right, I'll change the comment.